docs/java-rest/high-level/asyncsearch/submit.asciidoc

```diff
@@ -0,0 +1,94 @@
+--
+:api: asyncsearch-submit
+:request: SubmitAsyncSearchRequest
+:response: AsyncSearchResponse
+--
+
+[role="xpack"]
+[id="{upid}-{api}"]
+=== Submit Async Search API
+
+[id="{upid}-{api}-request"]
+==== Request
+
+A +{request}+ allows to submit an asynchronous search task to
+the cluster. Required arguments are the `SearchSourceBuilder` defining
+the search and the target indices:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request]
+--------------------------------------------------
+<1> The definition of the search to run
+<2> The target indices for the search
+
+==== Optional arguments
+The following arguments can optionally be provided:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request-arguments]
+--------------------------------------------------
+<1> The minimum time that the request should wait before
+returning a partial result (defaults to 1 second).
+<2> The expiration time of the result (defaults to 5 days).
+<3> Whether the resources should be kept after completion or failure
```

Comment:
s/resources/results ? I find this description confusing, shall we add what we have in the javadocs or rest spec?